it/controllers/business/BusinessWebsiteAddressControllerISpec.scala

```diff
@@ -0,0 +1,104 @@
+
+
+package controllers.business
+
+import itutil.ControllerISpec
+import models.BusinessContact
+import models.api.EligibilitySubmissionData
+import org.jsoup.Jsoup
+import play.api.libs.json.Format
+import play.api.libs.ws.WSResponse
+import play.api.test.Helpers._
+
+import scala.concurrent.Future
+
+class BusinessWebsiteAddressControllerISpec extends ControllerISpec {
+
+  val url: String = controllers.business.routes.BusinessWebsiteAddressController.show.url
+  val businessWebsiteAddress = "https://www.example.com"
+
+  val s4lData: BusinessContact = BusinessContact(
+    website = Some(businessWebsiteAddress)
+  )
+
+  s"GET $url" should {
+    "show the view correctly" in new Setup {
+      implicit val format: Format[BusinessContact] = BusinessContact.apiFormat
+      given()
+        .user.isAuthorised()
+        .s4lContainer[BusinessContact].isEmpty
+        .registrationApi.getSection[BusinessContact](Some(validBusinessContactDetails))
+        .registrationApi.getSection[EligibilitySubmissionData](Some(testEligibilitySubmissionData))
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: WSResponse = await(buildClient(url).get)
+      response.status mustBe OK
+    }
+
+    "return OK with prepopulated data" in new Setup {
+      given()
+        .user.isAuthorised()
+        .s4lContainer[BusinessContact].contains(s4lData)
+        .registrationApi.getSection[EligibilitySubmissionData](Some(testEligibilitySubmissionData))
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: Future[WSResponse] = buildClient(url).get()
+      whenReady(response) { res =>
+        res.status mustBe OK
+        Jsoup.parse(res.body).getElementById("businessWebsiteAddress").attr("value") mustBe businessWebsiteAddress
+      }
+    }
+  }
+
+  s"POST $url" when {
```

Comment:
hey! maybe ideal to also capture a scenario for failure test case, invalid regex leading in 400. thanks!